packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -181,11 +181,25 @@ Future<void> _writeCodeSizeAnalysis(BuildInfo buildInfo, SizeAnalyzer? sizeAnaly
   if (buildInfo.codeSizeDirectory == null || sizeAnalyzer == null) {
     return;
   }
-  final String arch = DarwinArch.x86_64.name;
-  final File aotSnapshot = globals.fs.directory(buildInfo.codeSizeDirectory)
-    .childFile('snapshot.$arch.json');
-  final File precompilerTrace = globals.fs.directory(buildInfo.codeSizeDirectory)
-    .childFile('trace.$arch.json');
+  final File? aotSnapshot = DarwinArch.values.map<File?>((DarwinArch arch) {
+    return globals.fs.directory(buildInfo.codeSizeDirectory).childFile('snapshot.${arch.name}.json');
+    // Pick the first if there are multiple for simplicity
```

Comment:
This could makes the code size output inaccurate[^1] depending on what binary the end user gets, right? I'm guessing us just picking the first one we find is _good enough_ because 

  1) This isn't a critical feature that's worth a bunch of extra logic.
  2) Most users of this (I recklessly assume) are only interested in _significant_ changes in code size (e.g. some new dependency pulls in 5MB of code when compiled). Because of this, outputting additional information per architecture is noisy at best and confusing at worst.

[^1] Maybe "misleading by omission" is more accurate.

Reply:
Right, so although we *might* have generated multiple aot snapshots (if Xcode was configured to build with multiple architectures), I think most users are only concerned about the relative binary size cost of the different dart libraries in their snapshot. And even if the arm64 and x64 snapshots were significantly different in size, they would be essentially comparing optimizations in dart compiler backends.

There may be a future reason why we users want/need access to every analysis json that we generated, but that would require further refactoring of tool code.